From 3c4024e350fbb75ff37699372e703408c3ad6ebd Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Thu, 19 Mar 2020 08:54:27 +0800
Subject: [PATCH] char: misc: Call misc_init() early when
 CONFIG_ROCKCHIP_THUNDER_BOOT=y

rfkill_init() depends on misc_init() to init misc_class.
But rfkill_init() and misc_init() are both in the same
subsys_initcall level.

Fix this by move misc_init() to the arch_initcall_sync level.

Change-Id: I7266c2787c04a42886816b0ef11f3ed968ebf557
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/char/misc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/char/misc.c b/drivers/char/misc.c
index 53cfe574d8d4..34c8eda52164 100644
--- a/drivers/char/misc.c
+++ b/drivers/char/misc.c
@@ -289,4 +289,8 @@ static int __init misc_init(void)
 		remove_proc_entry("misc", NULL);
 	return err;
 }
+#ifdef CONFIG_ROCKCHIP_THUNDER_BOOT
+arch_initcall_sync(misc_init);
+#else
 subsys_initcall(misc_init);
+#endif
-- 
2.35.3

